app/controllers/application_controller.rb

```diff
@@ -11,8 +11,14 @@ def append_info_to_payload(payload)
     payload[:user_agent] = request&.user_agent
   end
 
+  # With the ability to remove disclosure_check records we now look for
+  # a disclosure_report when we can't find the last created disclosure_check.
+  # This guarantees that any other disclosure_check records added by the user,
+  # will be available to that user.
   def current_disclosure_check
-    @_current_disclosure_check ||= DisclosureCheck.find_by_id(session[:disclosure_check_id])
```

Comment:
I'm feeling like there might be a simpler solution for this 😞 
The fact we need to look for one or the other, and set 2 IDs in the session feels itchy.

For now is ok but hopefully we can come up with a better solution.

Reply:
I'm hoping to rewrite it later, I think for now it's a nice middle step, then can focus on having DisclosureReport to be the starting point (and session) - I think!

Reply:
Cool, let's leave this for now. In the work I'm doing for the "change" functionality I'm also facing a similar problem where I need to refer to different disclosure_check record ID depending of the "change" link being clicked, so maybe there is some common ground between the 2 (remove and edit) and we can improve this.